src/Sarif.Multitool.Library/RewriteCommand.cs

```diff
@@ -56,13 +90,59 @@ public int Run(RewriteOptions options)
 
         private bool ValidateOptions(RewriteOptions rewriteOptions)
         {
-            bool valid = true;
+            if (!rewriteOptions.Validate())
+            {
+                return false;
+            }
 
-            valid &= rewriteOptions.Validate();
+            //  While this is returning true for inline cases, I think it's doing so for the wrong reasons.
+            //  TODO: validate whether "actualOutputPath" can be created.
+            if (!DriverUtilities.ReportWhetherOutputFileCanBeCreated(rewriteOptions.OutputFilePath, rewriteOptions.Force, _fileSystem))
+            {
+                return false;
+            }
 
-            valid &= DriverUtilities.ReportWhetherOutputFileCanBeCreated(rewriteOptions.OutputFilePath, rewriteOptions.Force, _fileSystem);
+            return true;
+        }
 
-            return valid;
+        //  TODO Move this into a separate class for better unit testing
+        private string SniffVersion(string sarifPath)
+        {
+            using (JsonTextReader reader = new JsonTextReader(new StreamReader(_fileSystem.FileOpenRead(sarifPath))))
+            {
+                while (reader.Read())
+                {
+                    if (reader.TokenType == JsonToken.PropertyName && ((string)reader.Value).Equals("version"))
+                    {
+                        reader.Read();
+                        return (string)reader.Value;
+                    }
+                }
+            }
+
+            return null;
+        }
+
+        private void TransformFileToVersionTwo(SingleFileOptionsBase options, string inputVersion, string outputFilePath)
```

Comment:
>void [](start = 16, length = 4)

can you return SarifLog instead of void?
in the if, you could return visitor.SarifLog

remove the else and return JsonConvert.... to SarifLog. 
With this, we won't need to save in disk

Reply:
i would change from options to the sarif string to facilitate, since we won't need to use the options

---
In reply to: [564740510](https://github.com/microsoft/sarif-sdk/pull/2252#discussion_r564740510) [](ancestors = 564740510)

Reply:
Done.

---
In reply to: [564742595](https://github.com/microsoft/sarif-sdk/pull/2252#discussion_r564742595) [](ancestors = 564742595,564740510)